src/zcbenchmarks.cpp

```diff
@@ -31,6 +31,31 @@
 #include "zcash/IncrementalMerkleTree.hpp"
 
 using namespace libzcash;
+// This method is based on Shutdown from init.cpp
+void shutdown_wallet()
+{
+    LogPrintf("%s: In progress...\n", __func__);
+    static CCriticalSection cs_Shutdown;
+    TRY_LOCK(cs_Shutdown, lockShutdown);
+    if (!lockShutdown)
+        throw new std::runtime_error("Failed to lock in shutdown_wallet");
```

Comment:
We should check what a `runtime_error` will do in the layers above here. If the node is shutting down, we don't want to crash it. Ideally the caller will just get an RPC error saying "The node is shutting down" or something.

Reply:
Why would that be the error message if the node failed to shut down? Anyways don't fully follow and haven't changed this

Reply:
This section isn't checking whether the node failed or not to shut down. It is trying to acquire a static lock, and if it fails then some other thread has the lock. However, now that I look again, I see this doesn't actually achieve what we'd want, because the lock here is going to be different to the lock in `init.cpp::Shutdown()`. So instead we should just call `ShutdownRequested()` instead of the locking code (ie. remove lines 38-39 and replace line 40).

Reply:
I just checked, and `runtime_error` should act fine within the RPC. But the error message should be changed to `The node is shutting down`

Reply:
Changed. Though, perhaps adding "by another process" in the end is clearer?